lib/resourcemerge/apps.go

```diff
@@ -19,6 +19,12 @@ func EnsureDeployment(modified *bool, existing *appsv1.Deployment, required apps
 		*modified = true
 		existing.Spec.Selector = required.Spec.Selector
 	}
+
+	if !equality.Semantic.DeepEqual(existing.Spec.Strategy, required.Spec.Strategy) {
+		*modified = true
+		existing.Spec.Strategy = required.Spec.Strategy
+	}
+
 	if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
```

Comment:
/hold

Can we move this so it doesn't sit between the two `spec.selector` blocks?  And do we also need similar `nil` checks?

Reply:
Good catch, hadn't realised I'd put it in the middle like that 🤦 

We don't need a nil check I don't think as the strategy isn't a pointer https://github.com/kubernetes/api/blob/9cb86d9d612512da24fae6761f7d6a2d1377dc0c/apps/v1/types.go#L309